### DISCUSSION TITLE
Advanced Mod Merge

### DISCUSSION THREAD

**Discussion by lxaphir** [bcssov/IronyModManager#203] [Q&A]
bcssov, what happened to the Advanced Mod Merge option in the latest IMM version?  I went to use it and it was gone.  I see compressed and basic.

**Comment by bcssov** [comment]
It was removed, I think I put it in the update notes. It was deprecated for a while now (ever since compress option was made available).

**Reply by lxaphir** [reply]
That's too bad, that was one thing that was working well for me.  Have you got some documentation that describes some of the atomic differences between what compression and advanced mod merges did?

**Reply by bcssov** [reply]
The wiki documentation git history can be used to see the differences in what advanced and what compress does. But in short:
1. Advanced behaved like basic merge with the sole difference it tried to deduplicate items in the game and attempted to mimic the game load behavior when rewriting scripts. This was error prone as irony relied on user made documentation on how the game loaded stuff and was difficult to maintain since each major update introduced some new quirks.
2. Compress is a lot simpler "merge" and regardless of updates it will always be up to date. It keeps mods separate as it mimics pre 2.4 stellaris behavior by simply zipping ali mods locally. So what it does is it will take x amount of mods and zip these x amount of mods by creating a local copy. Ex. 50 workshop mods are zipped, renamed and zipped as 50 local mods. Speed wise there is no difference between basic, advanced or merge.
